src/bicep/modules/firewall.bicep

```diff
@@ -27,11 +27,101 @@ resource firewallPolicy 'Microsoft.Network/firewallPolicies@2021-02-01' = {
   }
 }
 
+resource firewallAppRuleCollectionGroup 'Microsoft.Network/firewallPolicies/ruleCollectionGroups@2021-02-01' = {
+  name: '${firewallPolicyName}/DefaultApplicationRuleCollectionGroup'
+  dependsOn: [
+    firewallPolicy
+  ]
+  properties: {
+    priority: 300
+    ruleCollections: [
+      {
+        ruleCollectionType: 'FirewallPolicyFilterRuleCollection'
+        action: {
+          type: 'Allow'
+        }
+        rules: [
+          {
+            ruleType: 'ApplicationRule'
+            name: 'msftauth'
+            protocols: [
+              {
+                protocolType: 'Https'
+                port: 443
+              }
+            ]
+            fqdnTags: []
+            webCategories: []
+            targetFqdns: [
+              'aadcdn.msftauth.net'
+              'aadcdn.msauth.net'
+            ]
+            targetUrls: []
+            terminateTLS: false
+            sourceAddresses: [
+              '*'
+            ]
+            destinationAddresses: []
+            sourceIpGroups: []
+          }
+        ]
+        name: 'AzureAuth'
+        priority: 110
+      }
+    ]
+  }
+}
+
+resource firewallNetworkRuleCollectionGroup 'Microsoft.Network/firewallPolicies/ruleCollectionGroups@2021-02-01' = {
+  name: '${firewallPolicyName}/DefaultNetworkRuleCollectionGroup'
+  dependsOn: [
+    firewallPolicy
+    firewallAppRuleCollectionGroup
+  ]
+  properties: {
+    priority: 200
+    ruleCollections: [
```

Comment:
I think since this is such a complex object it makes sense to pass in as a `json(loadTextContent('./firewallRules/networkRules.json')` and document for users how to update this file